workflow/schemas/resources.schema.yaml

```diff
@@ -2,144 +2,170 @@ $schema: "http://json-schema.org/draft-04/schema#"
 description: snakemake configuration file
 type: object
 properties:
-  bwa_mem:
+  default_resources:
     type: object
+    description: default resource definitions
     properties:
-      threads:
-        type: integer
-        description: number of threads that will be used by the 'bwa_mem' rule
-      time:
-        type: string
-        description: max execution time for the 'bwa_mem' rule
       mem_mb:
         type: integer
-        description: memory used for 'bwa_mem'
+        description: memory in MB used per cpu
       mem_per_cpu:
         type: integer
         description: memory used per cpu for 'bwa_mem'
       partition:
         type: string
-        description: partition to use on the cluster for 'bwa_mem'
+        description: partition to use on cluster
+      time:
+        type: string
+        description: max execution time
+      threads:
+        type: integer
+        description:  number of threads to be available
+    required:
+      - threads
+      - time
+      - mem_mb
+      - mem_per_cpu
+      - partition
 
-  extract_reads:
+  bwa_mem:
     type: object
+    description: resource definitions for bwa
     properties:
-      threads:
-        type: integer
-        description: number of threads that will be used by the 'extract_reads' rule
-      time:
-        type: string
-        description: max execution time for the 'extract_reads' rule
       mem_mb:
         type: integer
-        description: memory used for 'extract_reads'
+        description: memory in MB used per cpu
       mem_per_cpu:
         type: integer
-        description: memory used per cpu for 'extract_reads'
+        description: memory used per cpu for 'bwa_mem'
       partition:
         type: string
-        description: partition to use on the cluster for 'extract_reads'
+        description: partition to use on cluster
+      time:
+        type: string
+        description: max execution time
+      threads:
+        type: integer
+        description:  number of threads to be available
 
-  mark_duplicates:
+  bwa_merge_bam:
     type: object
+    description: resource definitions for merging bwa bam files
     properties:
-      threads:
-        type: integer
-        description: number of threads that will be used by the 'mark_duplicates' rule
-      time:
-        type: string
-        description: max execution time for the 'mark_duplicates' rule
       mem_mb:
         type: integer
-        description: memory used for 'mark_duplicates'
+        description: memory in MB used per cpu
       mem_per_cpu:
         type: integer
-        description: memory used per cpu for 'mark_duplicates'
+        description: memory used per cpu for 'bwa_mem'
```

Comment:
this still points to `'bwa_mem'` and the others as well. Assuming it's copy-pasta. I would remove "for ..." so these descriptions are more general.

Reply:
removed bwa_mem part